### DISCUSSION TITLE
Zstandard data is incomplete on HEAD requests

### DISCUSSION THREAD

**Discussion by AlexMili** [encode/httpx#3408] [Potential Issue]
Tried to do a HEAD request with a server answering with `zstd` encoding by default but since the content is empty `ZStandardDecoder` is raising `DecodingError("Zstandard data is incomplete")` when calling `flush()`.

Tried to fix the problem by adding a check if `ret` is empty:
```python

    def flush(self) -> bytes:
        ret = self.decompressor.flush()  # note: this is a no-op
        # If content is empty, no need to wait for EOF
        if not self.decompressor.eof and len(ret) > 0:
            raise DecodingError("Zstandard data is incomplete")  # pragma: no cover
        return bytes(ret)
```

Don't know if there is an other way to avoid the error or fix it but sounds like a bug.

**Comment by lovelydinosaur** [comment]
Thanks seems viable. Can you point at a live URL to replicate?

**Reply by AlexMili** [reply]
You can try https://password-hashing.net/ or https://3proxy.org/.

Source: https://github.com/AlexMili/Reachable/issues/8
